### PR TITLE
regen/regcomp.pl - add more detail on what the script does and list its inputs properly

### PR DESCRIPTION
**Pull request by demerphq**
Previously we were only showing one of the inputs that generates regnodes.h, now we show all of them. Also add some paragraphs explaining a bit more detail what the files contain and how we process.

### CODE REVIEW

File: regen/regcomp.pl [hunk 1]
```diff
@@ -17,6 +18,34 @@
...
+# the regex engine to show things like which flags were set during
+# compilation. In some cases we transform the C code in the header files
+# into perl code which we execute to C<eval()> the contents. For instance
+# in a situation like this:
+#
+#   #define RXf_X 0x1   /* the X mode */
+#   #define RXf_Y 0x2   /* the Y mode */
+#   #define RXf_Z (X|Y) /* the Z mode */
+#
+# this script might end up eval()ing something like C<0x1> and then
```
Comment (jkeenan) [inline regen/regcomp.pl:40]: s/^this script/This script/

File: regen/regcomp.pl [hunk 2]
```diff
@@ -17,6 +18,34 @@
...
+# F<regcomp.sym> defines the opcodes and states used in the regex
+# engine, it also includes documentation on the opcodes. This script
+# parses those definitions out and turns them into typedefs, defines,
+# and data structures, and maybe even code which the regex engine can
+# use to operate.
+#
+# F<regexp.h> and op_reg_common.h contain defines C<RXf_xxx> and
+# C<PREGf_xxx> that are used in flags in our code. These defines are
+# parsed out and data structures are created to allow the debug mode of
+# the regex engine to show things like which flags were set during
```
Comment (jkeenan) [inline regen/regcomp.pl:31]: (At the risk of being pedantic) consider `show things such as which`.

Otherwise +1.

**Comment by tonycoz** [comment]
Looks good to me, with jkeenan's suggestions.

File: regen/regcomp.pl (hunk 1 above)
Comment (khwilliamson) [reply regen/regcomp.pl:40]: I disagree, `this script` is not beginning a sentence

**Review by khwilliamson — APPROVED** [review]

File: regen/regcomp.pl (hunk 1 above)
Comment (jkeenan) [reply regen/regcomp.pl:40]: > I disagree, `this script` is not beginning a sentence

Well, then, perhaps we could place ellipses at the end of the phrase above the code sample and at the beginning of the phrase after the code sample.  (This is a suggestion, not a requirement.)

Comment (jkeenan) [reply regen/regcomp.pl:40]: @khwilliamson, are you okay with my revised suggestion above?

**Comment by jkeenan** [comment]
In order to resolve this ticket, I'm accepting @khwilliamson's point in one instance and applying my grammatical correction in another instance.  Rebased on blead and merged into blead locally.  Pushed as 51135966bc.

**Comment by demerphq** [comment]
@jkeenan thanks for picking this up, I lost track of it.

File: regen/regcomp.pl (hunk 2 above)
Comment (demerphq) [reply regen/regcomp.pl:31]: Ah willdo in the future, sorry i missed this feedback!

**Pull request closed without merging**